usefulUtilities/toggleFlyButton/toggleFlyButton.js

```diff
@@ -0,0 +1,104 @@
+//
+//  toggleFlyButton.js
+//
+//  created by Zach Fox on 2019-01-10
+//  Copyright 2019 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+(function() { 
+    var _this;
+    var flyingEnabled;
+    var statusTextOverlayID;
+
+    var UPDATE_TEXT_DELAY_MS = 150;
+
+    var ToggleFlyButton = function() {
+        _this = this;
+    };
+
+    ToggleFlyButton.prototype = {
+        // Set up the entity ID, rez the status text overlay, and connect the
+        // HMD displayModeChanged signal
+        preload: function(entityID){
+            _this.entityID = entityID;
+            _this.rezStatusTextOverlay();
+            HMD.displayModeChanged.connect(_this.updateStatusText);
+        },
+        
+        // When the script goes down, delete the status text overlay (if it exists),
+        // and disconnect our signal
+        unload: function() {
+            if (statusTextOverlayID) {
+                Overlays.deleteOverlay(statusTextOverlayID);
+            }
+            HMD.displayModeChanged.disconnect(_this.updateStatusText);
+        },
+
+        // When the user uses the mouse to click on the button...
+        clickDownOnEntity: function(entityID, mouseEvent) {
+            if (mouseEvent.isLeftButton) {
+                _this.toggleFlyingEnabled();
+            }
+        },
+        
+        // When the user uses hand controller lasers to click on the button...
+        startNearTrigger: function() {
+            _this.toggleFlyingEnabled();
+        },
+        
+        // Create the status text overlay that'll show Flying Preference
+        // status.
+        rezStatusTextOverlay: function() {
+            statusTextOverlayID = Overlays.addOverlay("text3d", {
+                "parentID": _this.entityID,

```

Comment:
not a big one, but just curious why the keys are strings

Reply:
no difference in JS; doesn't matter. i mostly use strings for clarity and because that's (IIRC) what our JSON exporter does.